### PR TITLE
Update tutorial

### PR DESCRIPTION
**Pull request by Steven-Palayew**
Update tutorial to be a cleaned up, self contained Google Colab notebook

### CODE REVIEW

File: Tutorial.ipynb [hunk 1]
```diff
@@ -1,1105 +1,532 @@
...
+          "text": [
+            "/content/pathway-abstract-classifier/pathway-abstract-classifier\n",
+            "archive  dev-requirements.txt  requirements.txt  Tutorial.ipynb\n",
+            "Data\t README.md\t       tests\n"
+          ]
+        }
+      ],
+      "source": [
+        "%cd pathway-abstract-classifier\n",
+        "!ls\n",
```
Comment (JohnGiorgi) [inline Tutorial.ipynb:68]:
```suggestion
```

Looks like this is a debugging statement and was kept by accident

File: Tutorial.ipynb [hunk 2]
```diff
@@ -1,1105 +1,532 @@
...
-     "end_time": "2022-04-19T18:37:32.735337",
-     "exception": false,
-     "start_time": "2022-04-19T18:37:32.720200",
-     "status": "completed"
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c0ad5e77"
+      },
+      "source": [
+        "<h3> Below we import necessary libraries, turn on AMP (optional) and show two ways to load our model </h3>"
```
Comment (JohnGiorgi) [inline Tutorial.ipynb:81]: This is an aesthetic thing but all your text blocks appear to be headers? I would probably just make them plain text and use headers more sparingly to group content/sections.

File: Tutorial.ipynb [hunk 3]
```diff
@@ -1,1105 +1,532 @@
...
+        "#pre-process input data and make predictions (method 1 - using lists)\n",
+        "titles=df['title'].tolist()\n",
+        "abstracts=df['abstract'].tolist()\n",
+        "sep_token = predictor.preproc.get_tokenizer().sep_token\n",
+        "texts = [\" \".join([title, sep_token, abstract]) for title, abstract in zip(titles, abstracts)] \n",
+        "# Make predictions, checking how long they take to make (note- there are 1042 examples here this is using a (likely NVIDIA P100) GPU)\n",
+        "%time predictions=predictor.predict((texts))\n",
+        "\n",
+        "#pre-process input data and make predictions (method 2 - using dataframes, hard coded SEP token)\n",
+        "#texts=(df['title'].str.strip() + ' [SEP] ' + df['abstract'].str.strip()).to_numpy()\n",
```
Comment (JohnGiorgi) [inline Tutorial.ipynb:159]: There's lots of commented-out code throughout the notebook. If these lines are neccecary, they should be uncommented. If it is not, they should be removed.

File: Tutorial.ipynb [hunk 4]
```diff
@@ -1,1105 +1,532 @@
...
+        "# Unzip and read in validation data\n",
+        "import shutil\n",
+        "shutil.unpack_archive('../pathway-abstract-classifier/Data/val_data.tsv.zip', '../pathway-abstract-classifier/Data')\n",
+        "\n",
+        "df=pd.read_csv('../pathway-abstract-classifier/Data/val_data.tsv', delimiter=\"\\t\")\n",
+        "\n",
+        "#pre-process input data and make predictions (method 1 - using lists)\n",
+        "titles=df['title'].tolist()\n",
+        "abstracts=df['abstract'].tolist()\n",
+        "sep_token = predictor.preproc.get_tokenizer().sep_token\n",
```
Comment (JohnGiorgi) [inline Tutorial.ipynb:153]: I wouldn't flip between `sep_token` and `"[SEP]"`, that's confusing. I would use one or the other. If you feel this needs to be explained to the user that's OK. You could have a code cell that sets up `sep_token`. You could explain in plain English what you are doing and why and then use `sep_token` throughout (not the literal `"[SEP]"`).

**Review by JohnGiorgi — CHANGES_REQUESTED** [review]
Awesome! Looks good, I left quite a few comments but they are all small, wouldn't expect any of them to take more than 10 minutes to address. Also, it doesn't look like the environment uses a GPU by default? You can turn this on by selecting "GPU" under Runtime > Manage Runtime Type > Hardware Accelerator. This will save info to the notebook so that the next time someone opens it it will use a GPU.

File: Tutorial.ipynb (hunk 1 above)
Comment (Steven-Palayew) [reply Tutorial.ipynb:68]: I kept it there for some transparency into filesystem the way kaggle does by default - but it's cleaner without and am happy to change if you prefer.

File: Tutorial.ipynb (hunk 3 above)
Comment (Steven-Palayew) [reply Tutorial.ipynb:158]: I do this to show alternate ways of doing the same thing - I think that's valuable?

Comment (JohnGiorgi) [reply Tutorial.ipynb:158]: I would still uncomment it then, possibly moving it to a new code cell below, e.g. method 1 and method 2 would be adjacent code cells. Commenting it makes it seem like code that was forgotten about

Comment (Steven-Palayew) [reply Tutorial.ipynb:158]: Okay - and so should I comment it in the adjacent cell?

Comment (Steven-Palayew) [reply Tutorial.ipynb:158]: Alternative option I'm leaning towards:
- Delete alternate loading method since there are no files in local directory 
- Delete method to concatenate input using dataframes and hardcoded [SEP] - but point out to user (in a comment) that this method is valid and used in the files in archive

Comment (JohnGiorgi) [reply Tutorial.ipynb:158]: I would definitely delete it if it doesn't actually run. I really think this notebook should just be something I can Runtime > Run all and have no hiccups. Also, data frames are probably overkill and introduced a lot of overhead for something as simple as our preprocessing.

**Comment by Steven-Palayew** [comment]
Just committed updated tutorial based on your suggestions. I'm a little conflicted as to whether my subheadings looked a bit better since they didn't blend in as much, but I'm happy with this if you are.

**Comment by JohnGiorgi** [comment]
> @JohnGiorgi Just committed updated tutorial based on your suggestions. I'm a little conflicted as to whether my subheadings looked a bit better since they didn't blend in as much, but I'm happy with this if you are.

Awesome, LGTM. Nice and short and sweet. Will you just set the colab notebook to use GPU? Mentioned how to do that in my first comment above — you can ignore this if you’ve already done it but I didn’t notice. Otherwise let’s merge.

**Review by JohnGiorgi — APPROVED** [review]

**Comment by Steven-Palayew** [comment]
@JohnGiorgi I did this - is that what you were referring to? 
image

**Comment by JohnGiorgi** [comment]
> @JohnGiorgi I did this - is that what you were referring to? 
> 
> image
> 
> 

Yeah exactly, and then you saved and uploaded the ipynb here?

**Comment by Steven-Palayew** [comment]
Yes
